lib/kitchen/driver/ec2.rb

```diff
@@ -289,7 +289,8 @@ def default_ami
       def update_username(state)
         # TODO: if the user explicitly specified the transport's default username,
         # do NOT overwrite it!
-        if instance.transport[:username] == instance.transport.class.defaults[:username]
+        if actual_platform &&
+            instance.transport[:username] == instance.transport.class.defaults[:username]
```

Comment:
okay it'd be good to get that information documented in the TODO so we don't have to re-discover it all again at some point in the future.  you might just rewrite the TODO more like "BUG:  the default is often non-nil (e.g. 'root') but if the user specifies 'root' explicitly then we will overwrite that value -- need a way to distinguish a default from a set value"

its amusing how every configuration system inevitably re-discovers this problem...